addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -1825,6 +1825,9 @@ const ColorpickerUserValueWidget = SelectUserValueWidget.extend({
             this.colorPaletteEl.remove();
             this.colorPaletteEl = newEl;
         }
+        if (this.colorPaletteEl.closest('.o_we_so_color_palette').getAttribute('data-css-property') === 'background-color') {
```

Comment:
This part is not really good: this widget should not be concerned by the `data-css-property` value. It should not even know it exists. If we wanted this condition, it should be a param/option given to the widget during initialization.

But do we want this condition? I think we want the custom colors for any colorpicker that is opened :+1: Was not that the case in 16.0. I am assuming this is not buggy in 16.0 since you target 17.0? Would be nice if your commit message mentions why it is not buggy in 16.0 which is still supported.

Reply:
This issue appeared in 17.0. The colorpicker for background was not getting the props required (i.e. getEditableCustomColors was not loaded). I will add this information in the commit.

This test was to ensure we only make this assignement to background colorpickers so that is there is a change between the getEditableCustomColors for text and getEditableCustomColors for background, this would not be an issue.

Reply:
> The colorpicker for background was not getting the props required (i.e. getEditableCustomColors was not loaded). I will add this information in the commit.

That information is not really required in the commit message, we can see that by reading your diff. What is more relevant is "why was not it buggy in 16.0". In general, for any bug fix, we try to link the commit that introduced the issue (that avoids me asking "why is not it buggy in the previous version?"). I guess here, it's some kind of OWL refactoring that introduced the `getEditableCustomColors` function? :shrug:).

> This test was to ensure we only make this assignement to background colorpickers so that is there is a change between the getEditableCustomColors for text and getEditableCustomColors for background, this would not be an issue.

I don't understand :thinking: